realtime/utils.js

```diff
@@ -408,16 +408,29 @@ function attachAspectSubject(sample, subjectModel, aspectModel) {
     sample.subject ? sample.subject : subjectModel.findOne(subOpts),
   ])
   .then((response) => {
-    let asp = response[0];
-    let sub = response[1];
-    asp = asp.get ? asp.get() : asp;
+    const ASPECT_INDEX = 0;
+    const SUBJECT_INDEX = 1;
+    let asp = response[ASPECT_INDEX];
+    let sub = response[SUBJECT_INDEX];
+
+    if (!sub) {
+      /*
+       throw error when trying to attach the subject to a sample and
```

Comment:
this comment seems redundant since the error message is saying the same thing

Reply:
Awesome! DOne